packages/js/src/settings/helpers/search.js

```diff
@@ -232,9 +237,11 @@ export const createTaxonomySearchIndex = ( { name, label, route } ) => ( {
 /**
  * @param {Object} postTypes The post types.
  * @param {Object} taxonomies The taxonomies.
+ * @param {Object} options The options.
+ * @param {string} options.userLocale The user locale string.
  * @returns {Object} The search index.
  */
-export const createSearchIndex = ( postTypes, taxonomies ) => ( {
+export const createSearchIndex = ( postTypes, taxonomies, { userLocale = {} } = {} ) => ( {
```

Comment:
A `userLocale` should be a string (or array of strings if we use the `toLocaleLowerCase`API).
```suggestion
export const createSearchIndex = ( postTypes, taxonomies, { userLocale } = {} ) => ( {
```